src/pages/messages/components/NviCorrectionList.tsx

```diff
@@ -1,14 +1,76 @@
+import { Typography } from '@mui/material';
+import { useQuery } from '@tanstack/react-query';
+import { ParseKeys } from 'i18next';
 import { useTranslation } from 'react-i18next';
+import { useLocation } from 'react-router-dom';
+import { fetchResults, FetchResultsParams, ResultParam } from '../../../api/searchApi';
+import { allPublicationInstanceTypes } from '../../../types/publicationFieldNames';
+import { ROWS_PER_PAGE_OPTIONS } from '../../../utils/constants';
+import { nviApplicableTypes } from '../../../utils/registration-helpers';
+import { ScientificValueLevels } from '../../search/advanced_search/ScientificValueFilter';
+import { RegistrationSearch } from '../../search/registration_search/RegistrationSearch';
+
+export type CorrectionListId = '1' | '2';
+
+type CorrectionListSearchConfig = {
+  [key in CorrectionListId]: {
+    i18nKey: ParseKeys;
+    searchConfig: FetchResultsParams;
+  };
+};
+
+const correctionListConfig: CorrectionListSearchConfig = {
+  '1': {
+    i18nKey: 'tasks.nvi.correction_list_type.applicable_category_in_non_applicable_channel',
+    searchConfig: { categoryShould: nviApplicableTypes, scientificValue: ScientificValueLevels.LevelZero },
+  },
+  '2': {
+    i18nKey: 'tasks.nvi.correction_list_type.non_applicable_category_in_applicable_channel',
+    searchConfig: {
+      categoryShould: allPublicationInstanceTypes.filter((type) => !nviApplicableTypes.includes(type)),
+      scientificValue: [ScientificValueLevels.LevelOne, ScientificValueLevels.LevelTwo].join(','),
+    },
+  },
+};
+
+export const nviCorrectionListQueryKey = 'list';
 
 export const NviCorrectionList = () => {
   const { t } = useTranslation();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const listId = searchParams.get(nviCorrectionListQueryKey) as CorrectionListId | null;
+  const listConfig = listId && correctionListConfig[listId];
+
+  const fetchParams: FetchResultsParams = {
+    ...listConfig?.searchConfig,
+    from: Number(searchParams.get(ResultParam.From) ?? 0),
+    results: Number(searchParams.get(ResultParam.Results) ?? ROWS_PER_PAGE_OPTIONS[0]),
+  };
```

Comment:
todo: mangler støtte for sortering

Reply:
trenger også filter på årstall. Vi er bare interessert i nåværende år og forrige år i første omgang